packages/ra-core/src/reducer/admin/resource/list/validity.ts

```diff
@@ -0,0 +1,40 @@
+import { Reducer } from 'redux';
+import { FETCH_END } from '../../../../actions';
+import { GET_LIST } from '../../../../core';
+
+interface ValidityRegistry {
+    [key: string]: Date;
+}
+
+const initialState = {};
+
+const validityReducer: Reducer<ValidityRegistry> = (
```

Comment:
I think this would be better grouped with idsForQuery and totalForQuery to avoid having the same serialized query repeated three times though the state.